### ISSUE TITLE
improving performance of small-array quantity creation

### ISSUE THREAD

**Issue by thisch** [yt-project/unyt#110]
* unyt version:
0.2.3
* Python version:
cpython 3.7
* Operating System: GNU/Linux 

### Description

In the company I work with we create a lot of quantities and we care about performance. I've noticed that depending on how a quantity is created, creation-times may differ by a factor of 2. Currently we use python-quantities but we would like to switch to a different quantities library.

Since we usually create quantities using the ``X * pq.Y`` expression, we would benefit if ``3* unyt.mm`` were as fast as ``unyt.array.unyt_array(3, unyt.mm) `` ;)  Is this possible?

```
Python 3.7.3 (default, Aug 20 2019, 17:04:43) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.8.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import unyt

In [2]: %timeit unyt.array.unyt_array(3, unyt.mm)                                                                                                                                            
2.44 µs ± 14.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [3]: %timeit 3 * unyt.mm                                                                                                                                                                  
4.96 µs ± 36.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

As a reference here are the numbers for python-quantites
```
In [7]: import quantities as pq                                                                                                                                                              

In [8]: %timeit 3 * pq.mm                                                                                                                                                                    
7.77 µs ± 167 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [9]: %timeit pq.Quantity(3, pq.mm)                                                                                                                                                       
6.12 µs ± 82 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

**Comment by ngoldbaum** [comment]
Making it just as fast will probably be hard as we have to check whether the thing we're multiplying the unit by is another unit, a number, or an array.

We can certainly try to make it faster though.

**Comment by ngoldbaum** [comment]
I think I was able to close a substantial portion of the perfrmance gap you noticed:

```
In [2]: %timeit 3 * u.g
3.01 µs ± 31 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)

In [3]: %timeit u.array.unyt_array(3, u.g)
2.37 µs ± 66.9 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

See #111

**Comment by ngoldbaum** [comment]
Closed by #111
